include/deal.II/cgal/intersections.h

```diff
@@ -28,6 +28,18 @@
 DEAL_II_NAMESPACE_OPEN
 namespace CGALWrappers
 {
+  template <int spacedim, int n_vertices>
+  void
+  transform_to_cgal_vertex_order(
+    std::array<Point<spacedim>, n_vertices> &vertices);
+
+  template <int dim0, int dim1, int spacedim>
+  std::vector<std::array<Point<spacedim>, dim1 + 1>>
+  compute_intersection_of_cells(
+    const std::array<Point<spacedim>, int(std::pow(2, dim0))> &vertices0,
+    const std::array<Point<spacedim>, int(std::pow(2, dim1))> &vertices1,
```

Comment:
How did that work before? I think in the iterator version Hex, Quad and Lines were asserted.

Reply:
I see what you are saying. Just using n_components could get problematic since without the dimension it is not possible to derive the element type with the number of vertices. So probably one has to add dim and n_components.